pkg/reconciler/events/event.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2021 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package events
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	"knative.dev/pkg/apis"
+)
+
+const (
+	// Emit events if user provides enable-eventlistener-events: "true"
+	EnableEventListenerEvents = "enable-eventlistener-events"
+	// TriggerProcessingStarted is sent for Sink triggers received which
+	// are being processed
+	TriggerProcessingStartedV1 = "triggers.tekton.eventlistener.triggerprocessing.started.v1"
+	// TaskRunSuccessfulEventV1 is sent for Sink Triggers when a trigger is succssful
+	TriggerProcessingSuccessfulV1 = "triggers.tekton.eventlistener.triggerprocessing.successful.v1"
+	// TriggerProcessingFailedEventV1 is sent for Sink Triggers when we fail to process trigger
+	TriggerProcessingFailedEventV1 = "triggers.tekton.eventlistener.triggerprocessing.failed.v1"
+)
+
+// Emit emits events for object
+// Supported events are k8s events.
+func Emit(recorder record.EventRecorder, afterCondition *apis.Condition, object runtime.Object, err error) {
+	sendKubernetesEvents(recorder, afterCondition, object, err)
+}
+
+func sendKubernetesEvents(c record.EventRecorder, afterCondition *apis.Condition, object runtime.Object, err error) {
```

Comment:
I'm still not sure if we need to rely on the status of the EventListener for emitting events in Triggers - if the EL is not ready, we are probably not going to receive the event at all i.e. the server is not up or something. 

IMO it will be clearer emitEvents just sends the eventType (Successful, Failed etc.) as a param instead of the afterCondition.

Reply:
We had a scenario with `custom-resource` example as part of `0.16.0` Triggers release we had an issue  https://github.com/tektoncd/triggers/issues/1219 related to `metadata.resourceVersion` as part of Knative and because of that EL status was not **True** but pods were up and running so that is the reason i considered this way of using EL status in order to cover such scenarios also

Reply:
But agree with you that the current PR focuses on events for Trigger process so we no need to consider eventlistener status instead we can send the type as you suggested

I have updated PR PTAL